File: oop/heritage/CPP/exercice4/src/Agriculteur.cpp
#include "Agriculteur.hpp"

Agriculteur::Agriculteur() : Personne(), metier("") {}

Agriculteur::Agriculteur(const std::string &prenom) : Personne(prenom), metier("Agriculteur") {}

void Agriculteur::affiche() const
{
    Personne::affiche();
    std::cout << " le " << metier;
}

File: oop/heritage/Java/exercice4/src/Agriculteur.java


public class Agriculteur extends Personne {
    private String metier;

    public Agriculteur() {
        super();
    }

    public Agriculteur(String prenom) {
        super(prenom);
        metier = "Agriculteur";
    }

    @Override
    public void affiche() {
        super.affiche();
        System.out.print(" le " + this.metier);
    }
}
